.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,67 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, release-* ]
+  pull_request:
```

Comment:
Thanks for adding this @caniszczyk! I hadn’t seen this before and looks interesting. Noticed this becomes our longest running presubmit job, do you know if runtime reduces after first analysis of the base branch?

Reply:
If we are having issues with parallel jobs I can ask for more from GHA.

In regards to runtime, not sure if it gets faster, I fixed it to only look at the master branch tho now, if it finds anything maintainers will only see it here: https://github.com/crossplane/crossplane/security/code-scanning

Reply:
Hmm I was thinking maybe we only run it on merges (i.e. `push` in GH action parlance) rather than on each PR, but still run it against both the main branch and release branches. The issue is not parallelism with the jobs, but rather that the CodeQL job runs by itself longer than every other job that is running in parallel.